scip-java/src/main/scala/com/sourcegraph/scip_java/ScipPrinters.scala

```diff
@@ -0,0 +1,199 @@
+                  syntheticDefinition = Some(syntheticDefinition)
+                )
+              }
+          }
+        }
+      }
+    out.toString()
+  }
+
+  private def mopedPosition(input: Input, occ: Scip.Occurrence): Position = {
+    if (occ.getRangeCount == 3)
+      Position.range(
+        input,
+        occ.getRange(0),
+        occ.getRange(1),
+        occ.getRange(0),
+        occ.getRange(2)
+      )
+    else if (occ.getRangeCount == 4)
+      Position.range(
+        input,
+        occ.getRange(0),
+        occ.getRange(1),
+        occ.getRange(2),
+        occ.getRange(3)
+      )
+    else
+      throw new IllegalArgumentException(s"Invalid range: $occ")
+  }
+
+  private def formatOccurrence(
+      input: Input,
+      out: mutable.StringBuilder,
+      occ: Scip.Occurrence,
+      line: String,
+      symtab: Map[String, Scip.SymbolInformation],
+      comment: String,
+      syntheticDefinition: Option[Scip.SymbolInformation] = None
+  ): Unit = {
+    val pos = mopedPosition(input, occ)
+    val isMultiline = pos.startLine != pos.endLine
+    val width =
+      if (isMultiline) {
+        line.length - pos.startColumn - 1
+      } else {
+        math.max(1, pos.endColumn - pos.startColumn)
+      }
+
+    val isDefinition =
+      (occ.getSymbolRoles & SymbolRole.Definition.getNumber) > 0
+    val role =
+      if (syntheticDefinition.isDefined)
+        "synthetic_definition"
+      else if (isDefinition)
+        "definition"
+      else
+        "reference"
+    val indent =
+      if (pos.startColumn > comment.length)
+        " " * (pos.startColumn - comment.length)
+      else
+        ""
+    val caretCharacter =
+      if (syntheticDefinition.isDefined)
+        "_"
+      else
+        "^"
+    val carets =
+      if (pos.startColumn == 1)
+        caretCharacter * (width - 1)
+      else
+        caretCharacter * width
+
+    val symbol = syntheticDefinition.fold(occ.getSymbol)(_.getSymbol)
+    out
+      .append(comment)
+      .append(indent)
+      .append(carets)
+      .append(" ")
+      .append(role)
+      .append(" ")
+      .append(symbol)
+    if (isMultiline) {
+      out.append(s" ${pos.endLine - pos.startLine}:${pos.endColumn}")
+    }
+    out.append("\n")
+
+    syntheticDefinition.orElse(symtab.get(occ.getSymbol)) match {
+      case Some(info) if isDefinition =>
+        val prefix =
+          comment + (" " * indent.length) + (" " * carets.length) + " "
+        0.until(info.getDocumentationCount)
+          .foreach { n =>
+            val documentation = info.getDocumentation(n)
+            out
+              .append(prefix)
+              .append("documentation ")
+              .append(documentation.replace("\n", "\\n").replace("\t", "\\t"))
+              .append("\n")
+          }
+        0.until(info.getRelationshipsCount)
+          .foreach { n =>
+            val relationship = info.getRelationships(n)
+            out.append(prefix).append("relationship")
```

Comment:
Should we 

1. Sort relationships by symbol?
2. Sort documentation strings?

I can't reproduce the failures locally, but on CI it seems to be order related:

```diff
+//                      relationship is_reference is_implementation semanticdb maven . . com/airbnb/epoxy/BaseEpoxyTouchCallback#clearView().
 //                      relationship is_reference is_implementation semanticdb maven . . com/airbnb/epoxy/EpoxyTouchHelperCallback#clearView(+1).
 //                      relationship is_reference is_implementation semanticdb maven . . com/airbnb/epoxy/EpoxyTouchHelperCallback#clearView().
-//                      relationship is_reference is_implementation semanticdb maven . . com/airbnb/epoxy/BaseEpoxyTouchCallback#clearView().
 //                      ^ reference semanticdb maven . . com/airbnb/epoxy/EpoxyModelTouchCallback#[T]
```

Here's suggested patch (but this will require regenerating snapshots): 

```suggestion
        val docs =
          0.until(info.getDocumentationCount).map(info.getDocumentation).sorted

        docs.foreach { documentation =>
          out
            .append(prefix)
            .append("documentation ")
            .append(documentation.replace("\n", "\\n").replace("\t", "\\t"))
            .append("\n")
        }

        val relationships = 0
          .until(info.getRelationshipsCount())
          .map(info.getRelationships)
          .sortBy(_.getSymbol())

        relationships.foreach { relationship =>
          out.append(prefix).append("relationship")
```

Reply:
The failure was probably because I hadn't rebased on top of the scip-snapshots branch, which was failing with the same error. I've rebased now and will see if CI becomes green.